Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -466,131 +473,107 @@ void USpatialActorChannel::SetChannelActor(AActor* InActor)
 
 		// Inform USpatialInterop of this new actor channel/entity pairing
-		SpatialNetDriver->GetSpatialInterop()->AddActorChannel(ActorEntityId.ToSpatialEntityId(), this);
+		NetDriver->AddActorChannel(EntityId, this);
 	}
 }
 
-void USpatialActorChannel::SendReserveEntityIdRequest()
+FObjectReplicator& USpatialActorChannel::PreReceiveSpatialUpdate(UObject* TargetObject)
 {
-	USpatialNetConnection* SpatialConnection = Cast<USpatialNetConnection>(Connection);
-	check(SpatialConnection);
+	FNetworkGUID ObjectNetGUID = Connection->Driver->GuidCache->GetOrAssignNetGUID(TargetObject);
+	check(!ObjectNetGUID.IsDefault() && ObjectNetGUID.IsValid());
 
-	// Mark this channel as being responsible for creating this entity once we have an entity ID.
-	bCreatingNewEntity = true;
+	TargetObject->PreNetReceive();
+	FObjectReplicator& Replicator = FindOrCreateReplicator(TWeakObjectPtr<UObject>(TargetObject)).Get();
+	Replicator.RepLayout->InitShadowData(Replicator.RepState->StaticBuffer, TargetObject->GetClass(), (uint8*)TargetObject);
 
-	SpatialNetDriver->GetSpatialInterop()->SendReserveEntityIdRequest(this);
-}
-
-void USpatialActorChannel::PreReceiveSpatialUpdate(UObject* TargetObject)
-{
-	FNetworkGUID ObjectNetGUID = Connection->Driver->GuidCache->GetOrAssignNetGUID(TargetObject);
-	if (!ObjectNetGUID.IsDefault() && ObjectNetGUID.IsValid())
-	{
-		TargetObject->PreNetReceive();
-		FObjectReplicator& Replicator = FindOrCreateReplicator(TWeakObjectPtr<UObject>(TargetObject)).Get();
-		Replicator.RepLayout->InitShadowData(Replicator.RepState->StaticBuffer, TargetObject->GetClass(), (uint8*)TargetObject);
-	}
+	return Replicator;
 }
 
 void USpatialActorChannel::PostReceiveSpatialUpdate(UObject* TargetObject, const TArray<UProperty*>& RepNotifies)
 {
 	FNetworkGUID ObjectNetGUID = Connection->Driver->GuidCache->GetOrAssignNetGUID(TargetObject);
-	if (!ObjectNetGUID.IsDefault() && ObjectNetGUID.IsValid())
-	{
-		FObjectReplicator& Replicator = FindOrCreateReplicator(TWeakObjectPtr<UObject>(TargetObject)).Get();
-		TargetObject->PostNetReceive();
-		Replicator.RepNotifies = RepNotifies;
-		Replicator.CallRepNotifies(false);
-	}
-}
+	check(!ObjectNetGUID.IsDefault() && ObjectNetGUID.IsValid())
 
-void USpatialActorChannel::OnReserveEntityIdResponse(const worker::ReserveEntityIdResponseOp& Op)
-{
-	if (Op.StatusCode != worker::StatusCode::kSuccess)
-	{
-		UE_LOG(LogSpatialGDKActorChannel, Error, TEXT("Failed to reserve entity id. Reason: %s"), UTF8_TO_TCHAR(Op.Message.c_str()));
-		//todo: From now on, this actor channel will be useless. We need better error handling, or a retry mechanism here.
-		return;
-	}
-	UE_LOG(LogSpatialGDKActorChannel, Verbose, TEXT("Received entity id (%d) for: %s."), Op.EntityId.value_or(0), *Actor->GetName());
-	ActorEntityId = *Op.EntityId;
-	RegisterEntityId(ActorEntityId);
+	FObjectReplicator& Replicator = FindOrCreateReplicator(TWeakObjectPtr<UObject>(TargetObject)).Get();
+	TargetObject->PostNetReceive();
```

Comment:
Both here (and in Pre) we should check for `!IsPendingKill()`. (I know the original code wasn't doing it either.)